packages/components/src/button/style.scss

```diff
@@ -23,16 +23,10 @@
 	}
 
 	&[aria-expanded="true"],
-	&:hover {
+	&:hover:not(:disabled, [aria-disabled="true"]) {
 		color: $components-color-accent;
 	}
 
-	// Unset some hovers, instead of adding :not specificity.
```

Comment:
Reading this comment, there could be some unintended consequences across the editor in terms of rule specificity with these changes

Reply:
The CSS that was replaced by this was `&:not(:disabled):not([aria-disabled="true"]):not(.is-secondary):not(.is-primary):not(.is-tertiary):not(.is-link):hover`; I suspect that this was just intended for CSS simplification. It's always possible there's some unexpected consequences, but I don't think it's a significant concern.

Reply:
Reference: https://github.com/WordPress/gutenberg/pull/19344/files#diff-0688ad494526b3b921c4444dc8bc6f63efc4ad74c7c1123b061490cfc57f293a